src/arch/wasm/CodeGen.zig

```diff
@@ -990,44 +990,45 @@ fn addExtraAssumeCapacity(func: *CodeGen, extra: anytype) error{OutOfMemory}!u32
     return result;
 }
 
-/// Using a given `Type`, returns the corresponding type
+/// Using a given `Type`, returns the corresponding valtype for .auto callconv
+/// TODO: Merge this function with abi.classifyType, having different one for each callconv
```

Comment:
I made it to be geared towards _how to represent a type internally_, because I did not like this snippet `if (isByRef(...)) .i32 else typeToValtype(...)` present or erroneously missing throughout `CodeGen.zig`, where we want internal representation. I can keep changes to `typeToValtype` (but removing TODO), and modify `abi.classifyType` to return `union(enum)`, so code handling C callconv will not need to call `typeToValtype` at all.

Again the fact that `typeToValtype` returns `i64` for > 64 bits integers, is wrong because internally we store them as pointers to virtual stack, and only needed because of the `[2]Class` idea inside `abi.zig`.

Reply:
I will implement what I mean and push it, so you can review my idea in place, this I believe will result even in less hackiness inside `CodeGen.zig`.

Reply:
> I made it to be geared towards _how to represent a type internally_, because I did not like this snippet `if (isByRef(...)) .i32 else typeToValtype(...)` present or erroneously missing throughout `CodeGen.zig`, where we want internal representation. I can keep changes to `typeToValtype` (but removing TODO), and modify `abi.classifyType` to return `union(enum)`, so code handling C callconv will not need to call `typeToValtype` at all.
> 
> Again the fact that `typeToValtype` returns `.i64` for > 64 bits integers, is wrong because internally we store them as pointers to virtual stack, and only needed because of the `[2]Class` idea inside `abi.zig`.

We are on the same page. But my point is that the TODO is incorrect as those 2 functions should not be merged as they do different things.

Reply:
So you okay for me just removing "TODO" comment?

Reply:
Yes :)